src/librustc_typeck/check/dropck.rs

```diff
@@ -95,16 +94,20 @@ fn ensure_drop_params_and_item_params_correspond<'tcx>(
             }
             Err(_) => {
                 let item_span = tcx.def_span(self_type_did);
+                let def_kind = tcx.def_kind(self_type_did).unwrap();
                 struct_span_err!(
                     tcx.sess,
                     drop_impl_span,
                     E0366,
-                    "Implementations of Drop cannot be specialized"
+                    "`Drop` impls cannot be specialized"
                 )
                 .span_note(
                     item_span,
-                    "Use same sequence of generic type and region \
-                     parameters that is on the struct/enum definition",
+                    &format!(
+                        "use the same sequence of generic type and lifetime parameters \
```

Comment:
```suggestion
                        "use the same sequence of generic parameters \
```
We also have const parameters now.

Reply:
Fixed.